x-pack/plugins/siem/public/lib/compose/kibana_compose.ts

```diff
@@ -42,6 +42,7 @@ export function compose(): AppFrontendLibs {
     cache,
     link: ApolloLink.from([
       errorLink,
+      refetchOnErrorLink,
```

Comment:
Here should be really careful because of the order of the array, I will advise creating a function to return the array and create a unit test to make sure that the order stay the way we wanted it;

The order of the array is the way the `errorLink` and `refetchOnErrorLink` would be called meaning first `refetchOnErrorLink` and then `errorLink`. That's the way it works, I tried to change it and I got a surprise!!!

Reply:
yes, the order we listed here is from client-side to server-side, but the as we are handling server-side response, the flow is other way round, means that the error response is originally from server, goes through http link, retryOneTimeOnErrorLink, and finally errorLink (please find the diagram above in the description).
I've created `getLinks` and extracted it in `helpers.js` (as I had a dependency problem when importing `compose` function to the test ) to retrieve all these links and a unit test to make sure it alerts if the order is changed.